astronomer/providers/core/operators/external_trigger_dagrun.py

```diff
@@ -0,0 +1,417 @@
+                ).run_id
+
+            api.clear_dag_run(dag_id=self.trigger_dag_id, run_id=self.trigger_run_id)
+            dag_run = api.get_dag_run(dag_id=self.trigger_dag_id, run_id=self.trigger_run_id)
+
+        if dag_run is None:
+            raise RuntimeError("The dag_run should be set here!")
+        logging.warning(type(dag_run))
+        logging.warning(inspect.getmembers(dag_run))
+
+        self.defer(
+            timeout=self.execution_timeout,
+            trigger=ExternalDeploymentDagRunTrigger(
+                http_conn_id=self.http_conn_id,
+                method="GET",
+                endpoint=DAGRUN_ENDPOINT.format(dag_id=dag_run.dag_id, run_id=dag_run.run_id),
+                data=None,
+                headers=self.headers,
+                extra_options=self.extra_options,
+                poke_interval=self.poke_interval,
+                dag_id=dag_run.dag_id,
+                run_id=dag_run.run_id,
+            ),
+            method_name="execute_complete",
+        )
+
+    def execute_complete(self, context: Context, event: Any = None) -> None:
+        """
+        Callback for when the trigger fires - returns immediately.
+
+        Relies on trigger to throw an exception, otherwise it assumes execution was
+        successful.
+
+        :param context: Airflow context
+        :param event: Trigger Event
+        """
+        state = event.get("state")
+        dag_id = event.get("dag_id")
+        run_id = event.get("run_id")
+        execution_date = event.get("execution_date")
+        if state in self.allowed_states:
+            self.log.info(f"Dag Run Succeeded with response: {event}")
+            if dag_id:
+                context["ti"].xcom_push(key=XCOM_DAG_ID, value=dag_id)
+            if run_id:
+                context["ti"].xcom_push(key=XCOM_RUN_ID, value=run_id)
+            if execution_date:
+                context["ti"].xcom_push(key=XCOM_EXECUTION_DATE_ISO, value=execution_date)
+            return
+        raise AirflowException(f"Dag Run Failed with response: {event}")
+
+
+#######################################
+def parse_execution_date(execution_date: datetime | str) -> datetime:
+    """
+    Parse execution date to datetime object.
+
+    :param execution_date: Execution date
+    :return: Execution date as datetime object
+    """
+    if isinstance(execution_date, datetime):
+        return execution_date
+    elif isinstance(execution_date, str):
+        return cast(datetime, timezone.parse(execution_date))
+    else:
+        raise TypeError(
+            f"Expected str or datetime.datetime type for execution_date.Got {type(execution_date)}"
+        )
+
+
+def parse_dag_run(json_dict: dict[str, Any]) -> DagRun:
+    """
+    Parse JSON dictionary to DagRun object.
+
+    :param json_dict: Dictionary containing DagRun properties to parse
+    """
+    if (data_interval_start := json_dict.get("data_interval_start")) and (
```

Comment:
Sadly, the [setup.cfg](https://github.com/astronomer/astronomer-providers/blob/main/setup.cfg#L32) mentions `python>=3.7`. So the usage of walrus operators may not be supported on `3.7`

If we wish to continue using this, we might need to raise an exception while using the operator is Python version `<3.8` or refactor this bit here to also support `3.7 `